sqlglot/dialects/athena.py

```diff
@@ -20,12 +20,56 @@ def _generate_as_hive(expression: exp.Expression) -> bool:
         else:
             return expression.kind != "VIEW"  # CREATE VIEW is never Hive but CREATE SCHEMA etc is
 
-    elif isinstance(expression, exp.Alter) or isinstance(expression, exp.Drop):
-        return True  # all ALTER and DROP statements are Hive
+    # https://docs.aws.amazon.com/athena/latest/ug/ddl-reference.html
+    elif isinstance(expression, (exp.Alter, exp.Drop, exp.Describe)):
+        if isinstance(expression, exp.Drop) and expression.kind == "VIEW":
+            # DROP VIEW is Trino (I guess because CREATE VIEW is)
+            return False
+
+        # Everything else is Hive
+        return True
 
     return False
 
 
+def _location_property_sql(self: Athena.Generator, e: exp.LocationProperty):
+    # If table_type='iceberg', the LocationProperty is called 'location'
+    # Otherwise, it's called 'external_location'
+    # ref: https://docs.aws.amazon.com/athena/latest/ug/create-table-as.html
+
+    prop_name = "external_location"
+
+    if isinstance(e.parent, exp.Properties):
+        table_type_property = next(
+            (
+                p
+                for p in e.parent.expressions
+                if isinstance(p, exp.Property) and p.name == "table_type"
+            ),
+            None,
+        )
+        if (
+            table_type_property
+            and table_type_property.value
+            and table_type_property.value.name == "iceberg"
+        ):
+            prop_name = "location"
+
+    return f"{prop_name}={self.sql(e, 'this')}"
+
+
+def _alter_sql(self: Athena._HiveGenerator, e: exp.Alter):
```

Comment:
this is not really a transform, but it's fine, can you move this directly into the Generate and make it a proper subclass method instead of using transforms